### PR TITLE
Limit the amount of GitHub Actions time consumed by tests

### PR DESCRIPTION
**Pull request by fdel-car**
### What does it do?

Add `concurrency` and `paths-ignore` settings to limit the amount of time the GitHub Actions runner is used.
It should make so consecutive pushes on a PR will not all lead to a full run of tests.

Also, you can notice that I have removed the `push` trigger on `main` as I don't think it's needed since this branch should be protected and it would lead to unnecessary runs.

One thing that is worth mentioning is that GitHub Actions does not provide a long grace period when canceling runs, so if our `tests` workflow needs time to clean up stuff, then this PR shouldn't go through as is with this `concurrency` setting.

### Why is it needed?

Computation power and GitHub Actions time are wasted unnecessarily,

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -1,10 +1,14 @@
 name: 'Tests'
 
 on:
-  push:
```
Comment (alexandrebodin) [inline .github/workflows/tests.yml:4]: we'll have to enforce "update with main" before merge if we remove it. Can you put it in another PR so we can merge the rest while this point is discussed.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
-    branches:
-      - main
   pull_request:
+    paths-ignore:
+      - 'docs/**'
+      - '**/**.md'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_name }}
```
Comment (alexandrebodin) [inline .github/workflows/tests.yml:10]: Isn't this concurrency setting already scoped to the workflow?

**Review by alexandrebodin — COMMENTED** [review]
Very nice. It's a good 1st step before we split this further 👍

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -1,10 +1,14 @@
 name: 'Tests'
 
 on:
-  push:
-    branches:
-      - main
   pull_request:
+    paths-ignore:
+      - 'docs/**'
+      - '**/**.md'
```
Comment (gu-stav) [inline .github/workflows/tests.yml:7]: Could we also add `- '**/**.mdx'`?

File: .github/workflows/tests.yml (hunk 2 above)
Comment (fdel-car) [reply .github/workflows/tests.yml:10]: No, if we remove the workflow from that identifier then every other job from other workflows will get cancelled, you can see that [here](https://docs.github.com/en/actions/using-jobs/using-concurrency#example-only-cancel-in-progress-jobs-or-runs-for-the-current-workflow)

File: .github/workflows/tests.yml (hunk 3 above)
Comment (fdel-car) [reply .github/workflows/tests.yml]: Of course, I actually updated it to include what you've requested by updating this to `**.mdx?`. You can get more info [here](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet)

**Comment by fdel-car** [comment]
By the way, if you were wondering if it worked you can check [here](https://github.com/strapi/strapi/actions/runs/2926526480), which was caused by two successive commits that I made on this branch 😄

File: .github/workflows/tests.yml (hunk 2 above)
Comment (alexandrebodin) [reply .github/workflows/tests.yml:13]: Good to know 👍

**Review by alexandrebodin — APPROVED** [review]
